sched/semaphore/sem_destroy.c

```diff
@@ -72,10 +74,18 @@ int nxsem_destroy(FAR sem_t *sem)
    * leave the count unchanged but still return OK.
    */
 
-  if (sem->semcount >= 0)
+  do
     {
-      sem->semcount = 1;
```

Comment:
I think this PR should be split to two, one is replace semaphore count type to atomic, another one is fast mutex

Reply:
If fast mutexes are implemented the implementation should be in user space (i.e. libc) for it to benefit memory protected builds as well. There taking the lock (in our case, semaphore) is a very expensive operation via syscall.

Reply:
Yes, this patch only optimizes the mutex performance of the kernel, while the performance optimization of the userspace will be implemented in subsequent patches.